### PR TITLE
Merge latest openjdk

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Merge latest openjdk

Conflicts:
```
	src/jdk.jfr/share/classes/jdk/jfr/internal/consumer/ConstantMap.java
	test/hotspot/jtreg/compiler/c2/TestMinValueStrideLongCountedLoop.java
	test/hotspot/jtreg/compiler/rangechecks/TestArrayAccessCastIIAboveRC.java
	test/hotspot/jtreg/gc/stress/gclocker/TestGCLockerWithParallel.java
```

Modified:
```
	src/java.base/share/classes/java/lang/VirtualThread.java
	src/hotspot/share/include/jvm.h
```

The merging conflict error:
```
Auto-merging test/hotspot/jtreg/gc/stress/gclocker/TestGCLockerWithParallel.java
CONFLICT (rename/rename): test/hotspot/jtreg/gc/stress/gclocker/TestGCLockerWithParallel.java renamed to test/hotspot/jtreg/compiler/rangechecks/TestArrayAccessCastIIAboveRC.java in HEAD and to test/hotspot/jtreg/compiler/c2/TestMinValueStrideLongCountedLoop.java in e0a264692aec6713c27ef7f49558a5a28706339c.
```

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by keithc-ca** [comment]
This seems to have commits that aren't (yet?) on the master branch: How did that happen?

**Comment by keithc-ca** [comment]
https://github.com/ibmruntimes/openj9-openjdk-jdk/compare/d6db9a2d3948b861d4659a00faf00226e9b51511..e0a264692aec6713c27ef7f49558a5a28706339c

**Comment by JasonFengJ9** [comment]
> This seems to have commits that aren't (yet?) on the master branch: How did that happen?

This PR targets the `master` branch, not the usual `openj9-staging` one.

File: src/hotspot/share/include/jvm.h [hunk 1]
```diff
@@ -1152,7 +1152,10 @@ JNIEXPORT void JNICALL
...
 
 JNIEXPORT void JNICALL
-JVM_VirtualThreadHideFrames(JNIEnv* env, jobject vthread, jboolean hide);
+JVM_VirtualThreadHideFrames(JNIEnv* env, jclass clazz, jboolean hide);
+
+JNIEXPORT void JNICALL
+JVM_VirtualThreadDisableSuspend(JNIEnv* env, jclass clazz, jboolean enter);
 
 JNIEXPORT void JNICALL
 JVM_VirtualThreadDisableSuspend(JNIEnv* env, jobject vthread, jboolean enter);
```
Comment (keithc-ca) [inline src/hotspot/share/include/jvm.h:1161]: This should not be here (see lines 1157-1158).

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: src/java.base/share/classes/java/lang/VirtualThread.java [hunk 2]
```diff
@@ -1175,7 +1175,10 @@ private void setCarrierThread(Thread carrier) {
...
     @IntrinsicCandidate
     @JvmtiMountTransition
-    private native void notifyJvmtiHideFrames(boolean hide);
+    private static native void notifyJvmtiHideFrames(boolean hide);
+
+    @IntrinsicCandidate
+    private static native void notifyJvmtiDisableSuspend(boolean enter);
 
     @IntrinsicCandidate
     private native void notifyJvmtiDisableSuspend(boolean enter);
```
[the remark this answers is not in the document]
Comment (JasonFengJ9) [reply src/java.base/share/classes/java/lang/VirtualThread.java]: Rerun the script w/ the latest openjdk update, there was no merging conflict at `src/java.base/share/classes/java/lang/VirtualThread.java` and `src/hotspot/share/include/jvm.h`.

Replaced it with openjdk version.

File: src/hotspot/share/include/jvm.h (hunk 1 above)
Comment (JasonFengJ9) [reply src/hotspot/share/include/jvm.h]: Replace it w/ openjdk version.

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**